app/js/components/LabOrdersList.jsx

```diff
@@ -154,6 +177,50 @@ export class LabOrdersList extends PureComponent {
     dispatch(filtersAction.setLabOrdersListFilters(newFilters));
   }
 
+  async handleCancel(order) {
+    const {
+      currentProvider,
+      dispatch,
+      testOrderEncounterType,
+      sessionLocation,
+      testOrderEncounterRole,
+    } = this.props;
+    const cancelConfirmation = await swal("Are you sure you would like to cancel this order ?", {
+      buttons: {
+        YES: "YES",
+        NO: 'NO',
+      },
+    });
+    if (cancelConfirmation === "YES") {
+      const cancelledOrder = {
+        careSetting: order.careSetting,
+        concept: order.concept,
+        patient: order.patient.uuid,
+        encounter: null,
+        action: "DISCONTINUE",
+        orderer: currentProvider.uuid,
+        previousOrder: order.uuid,
+        type: order.type,
+        urgency: order.urgency,
+      };
+
+      const payload = {
```

Comment:
We don't want to create a whole new encounter,so  we should just be able to post the "discontinue" order to the orders endpoint.

Sorry, I should have noticed this before...  it looks like they creating a whole new encounter in the Order Entry module when cancelled an order... this is incorrect, and I will need to ticket fixing that.

It also looks like we don't have a "createOrder" method in orderRest.js, so we will need to add that... it should be able to follow the pattern of the createEncounter post in encounterRest.js

Reply:
Thanks a lot, @mogoodrich this works perfectly. I found out I don't even need all those encounter roles and types anymore. so would have to close out the other PR on react-components and add the createOrder in order rest.